cpp/source/deal.II/geometry.templates.h

```diff
@@ -313,26 +291,50 @@ void Geometry<dim>::fill_materials_map(
             material_database.get<std::string>("material_id"));
     std::string const material_name =
         material_database.get<std::string>("name");
-    _materials->emplace(material_name, material_ids);
+    auto const material_weight =
+        material_database.get<unsigned int>("weight", 0);
+    _materials->emplace(material_name,
+                        std::set<dealii::types::material_id>(
+                            material_ids.begin(), material_ids.end()));
+    _weights.emplace(material_name, material_weight);
+  }
+  _boundaries = std::make_shared<
+      std::unordered_map<std::string, std::set<dealii::types::boundary_id>>>();
+  int const n_boundaries = database->get<int>("boundaries");
+  for (int b = 0; b < n_boundaries; ++b)
+  {
+    boost::property_tree::ptree boundary_database =
+        database->get_child("boundary_" + std::to_string(b));
+    std::vector<dealii::types::boundary_id> const boundary_ids =
+        to_vector<dealii::types::boundary_id>(
+            boundary_database.get<std::string>("boundary_id"));
+    std::string const boundary_name =
+        boundary_database.get<std::string>("name");
+    _boundaries->emplace(boundary_name,
+                         std::set<dealii::types::boundary_id>(
+                             boundary_ids.begin(), boundary_ids.end()));
   }
 }
 
 template <int dim>
-void Geometry<dim>::fill_materials_map()
+void Geometry<dim>::fill_material_and_boundary_maps()
 {
-  _materials = std::make_shared<std::unordered_map<
-      std::string, std::vector<dealii::types::material_id>>>();
-  (*_materials)["anode"] = std::vector<dealii::types::material_id>(1, 0);
-  (*_materials)["separator"] = std::vector<dealii::types::material_id>(1, 1);
-  (*_materials)["cathode"] = std::vector<dealii::types::material_id>(1, 2);
-  (*_materials)["collector_anode"] =
-      std::vector<dealii::types::material_id>(1, 3);
-  (*_materials)["collector_cathode"] =
-      std::vector<dealii::types::material_id>(1, 4);
-  std::vector<dealii::types::material_id> coll_material_ids(2);
-  coll_material_ids[0] = 3;
-  coll_material_ids[1] = 4;
-  (*_materials)["collector"] = coll_material_ids;
+  _materials = std::make_shared<
+      std::unordered_map<std::string, std::set<dealii::types::material_id>>>(
+      std::initializer_list<
+          std::pair<std::string const, std::set<dealii::types::material_id>>>{
+          {"anode", std::set<dealii::types::material_id>{0}},
+          {"separator", std::set<dealii::types::material_id>{1}},
+          {"cathode", std::set<dealii::types::material_id>{2}},
+          {"collector_anode", std::set<dealii::types::material_id>{3}},
```

Comment:
I misunderstood the "these". `collector_anode` and `collector_cathode` are used in `Geometry<dim>::set_boundary_ids`.